legend-engine-xt-data-space-generation/src/test/java/org/finos/legend/engine/generation/TestDataSpaceAnalyticsArtifactGenerationExtension.java

```diff
@@ -0,0 +1,76 @@
+import org.finos.legend.engine.language.pure.compiler.Compiler;
+import org.finos.legend.engine.language.pure.compiler.toPureGraph.PureModel;
+import org.finos.legend.engine.language.pure.dsl.generation.extension.Artifact;
+import org.finos.legend.engine.language.pure.grammar.from.PureGrammarParser;
+import org.finos.legend.engine.protocol.pure.v1.model.context.PureModelContextData;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.dataSpace.DataSpace;
+import org.finos.legend.engine.shared.core.deployment.DeploymentMode;
+import org.finos.legend.engine.shared.core.deployment.DeploymentStateAndVersions;
+import org.finos.legend.pure.generated.Root_meta_pure_metamodel_dataSpace_DataSpace;
+import org.junit.Assert;
+import org.junit.Test;
+
+
+public class TestDataSpaceAnalyticsArtifactGenerationExtension
+{
+
+    private String getResourceAsString(String path)
+    {
+        try
+        {
+            URL infoURL = DeploymentStateAndVersions.class.getClassLoader().getResource(path);
+            if (infoURL != null)
+            {
+                java.util.Scanner scanner = new java.util.Scanner(infoURL.openStream()).useDelimiter("\\A");
+                return scanner.hasNext() ? scanner.next() : null;
+            }
+            return null;
+        }
+        catch (Exception e)
+        {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Test
+    public void testDataSpaceAnalyticsArtifactGenerationExtension()
+    {
+        String pureModelString = getResourceAsString("models/DataspaceModel.pure");
+        PureModelContextData pureModelContextData = PureGrammarParser.newInstance().parseModel(pureModelString);
+        PureModel pureModel = Compiler.compile(pureModelContextData, DeploymentMode.TEST, null);
+        Map<DataSpace, List<Artifact>> results = Maps.mutable.empty();
+        DataSpaceAnalyticsArtifactGenerationExtension extension = new DataSpaceAnalyticsArtifactGenerationExtension();
+        org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.PackageableElement packageableElement = pureModel.getPackageableElement("dataSpace::_FirmDataSpace");
+        Assert.assertTrue(packageableElement instanceof Root_meta_pure_metamodel_dataSpace_DataSpace);
+        Root_meta_pure_metamodel_dataSpace_DataSpace metamodelDatasapce = (Root_meta_pure_metamodel_dataSpace_DataSpace) packageableElement;
+        Assert.assertTrue(extension.canGenerate(metamodelDatasapce));
+        List<Artifact> outputs = extension.generate(packageableElement, pureModel, pureModelContextData, "vX_X_X");
+        Assert.assertEquals(1, outputs.size());
+        Artifact dataSpaceAnalyticsResult = outputs.get(0);
+        Assert.assertEquals(dataSpaceAnalyticsResult.format, "json");
+        Assert.assertEquals(dataSpaceAnalyticsResult.path, "AnalyticsResult.json");
+        Assert.assertNotNull(dataSpaceAnalyticsResult.content);
+        Assert.assertNotEquals("", dataSpaceAnalyticsResult.content);
```

Comment:
Don't we have some substantive expectation of what the content will be?

Reply:
I think we should just assert the content is non-empty, @MauricioUyaguari and I discussed about this and we think changing the generator would fail this test, which doesn't make much sense since this is testing the generic mechanism of the extension, not pertaining to any specific plugin. What do you think?

Reply:
I'm not sure what you mean by "not pertaining to any specific plugin". This is a test specifically for the DataSpace plugin, isn't it?

At the very least, you should make sure that the generated content can be used for its intended purpose. Presumably, it needs to be valid JSON that can be parsed into a certain target class. That should be easy enough to test, and not brittle.

Reply:
Makes sense, updated!